spec/facts_spec.rb

```diff
@@ -118,12 +118,18 @@
         expect(content['systemd_internal_services']).to be_nil
       end
 
-      # those are used in newer rspec-puppet-facts versions
-      # by ensuring those facts exists, we can drop a huge amount of logic in rspec-puppet-facts
-      it 'contains the mandatory OS facts' do
-        expect(content['os']['release']['major']).to not_be_nil.and not_be_empty
-        expect(content['os']['name']).to not_be_nil.and not_be_empty
-        expect(content['os']['hardware']).to not_be_nil.and not_be_empty
```

Comment:
can you maybe keep those checks? we rely on those in rspec-puppet-facts and assume they exist. And there were situations in the past where factsets were provided for operating systems that aren't 100% supported and those facts were missing or null.

Reply:
Good catch, I think that the initial code was looking for the legacy facts, and after rebasing I misread the conflicts that changed them to os.xxx facts and removed them by mistake.

Reintroduced, but I kept the check that ensure legacy facts are not found.